Spec/mastermind_spec.rb

```diff
@@ -101,4 +101,17 @@
   it "returns invalid color message" do
     expect { response.color_error }.to output(/Invalid colors/).to_stdout
   end
+
+  it "validates choice of length" do 
+    expect(response.valid_length?).to be false
+  end
+
+  it "returns invalid length message" do 
+    expect { response.length_error }.to output(/Invalid length/).to_stdout
+  end
+
+  it "quits the game if \"quit\" is entered" do
+    response = Response.new("quit")
+    expect{ response.leave }.to output(/Are you sure you want to quit?/).to_stdout
+  end
 end
```

Comment:
These last two tests are pretty brittle. I wouldn't recommend testing output to the console as a reliable test of behavior. I would even recommend just cutting these last two tests out. `leave` and `restart` aren't doing a lot except for loading and exiting the application which we know Ruby knows how to do with `load` and `abort`. Maybe this is an example of over-testing.